studiopress/pattern-manager
# Fix preview width when searching or filtering patterns

Related to the work in https://github.com/studiopress/pattern-manager/pull/109 and https://github.com/studiopress/pattern-manager/pull/95 — preview widths are sometimes stale when filtering by clicking category options or searching in the patterns view:

![pm-preview-sizing-issue](https://user-images.githubusercontent.com/108079556/225124528-4819f52e-c878-47e5-94f3-35689438c5cf.gif)

The width expands when it is the only filtered option, but remains at full-width when it should be resized to accommodate a second result.

---

This PR expands on and replaces the `useWindowSize` hook in https://github.com/studiopress/pattern-manager/pull/109 to allow other dependencies (such as `filteredPatterns`/`themePatterns`) to also trigger a re-render:

![pm-preview-sizing-fix](https://user-images.githubusercontent.com/108079556/225125163-59422744-d4fa-4ff6-bab9-dcf44f9c831d.gif)

Now, regardless of how the user searches, clicks through categories, or resizes the window, we can also be sure a fresh grid of pattern previews with correct dimensions is displayed.

---

### How to test

1. Checkout the branch
2. Filter the patterns by searching and/or clicking various categories
3. Resize your browser window
4. The previews should remain at the expected scale throughout

---

### Notes
Not that this was expected to change, but even with the extra rendering introduced by these changes, Lighthouse scores remain screaming fast. Care was taken to ensure this does not cause an unneeded re-render on initial load:

![Screenshot 2023-03-14 at 3 13 50 PM](https://user-images.githubusercontent.com/108079556/225126377-92093dd6-6bdf-4b42-b83f-2c624b05478c.png)
